### ISSUE TITLE
Bug: `eslint-plugin` -> `configs[flat/*]` has incorrect types

### ISSUE THREAD

**Issue by abrahamguo** [typescript-eslint/typescript-eslint#12791]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.
- [x] If I used AI assistance to prepare this issue, I have reviewed and followed the [AI Contribution Policy](https://typescript-eslint.io/contributing/ai-policy/).

### Relevant Package

eslint-plugin

### Repro Code

```TypeScript
import tseslint from '@typescript-eslint/eslint-plugin';

tseslint.configs['flat/base'].plugins?.join(', ');
```

### ESLint Config

```javascript

```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

The code should emit a tsc error, since it crashes at runtime:

```
TypeError: tseslint.configs.flat/base.plugins?.join is not a function
```

### Actual Result

The code emits zero tsc errors, even though it crashes at runtime.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `8.68.0` |

**Comment by bradzacher** [comment]
This is expected because it's a hacky way to expose both legacy and flat configs.

This does not form part of our public API and you should instead be accessing the flat configs via the `typescript-eslint` package.

**Comment by kirkwaiblinger** [comment]
Yeah this is basically one of the warts we had to accept in https://github.com/typescript-eslint/typescript-eslint/pull/10973.

**Comment by kirkwaiblinger** [comment]
It might be possible to enumerate the actual classic configs in 

https://github.com/typescript-eslint/typescript-eslint/blob/8ef5f4bffcf0904a28ada92a224a378d309407a4/packages/eslint-plugin/index.d.ts#L8-L13 

rather than use an index signature, but seems not worth changing it.
